internal/services/bot/bot_resource_test.go

```diff
@@ -0,0 +1,190 @@
+package bot_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/bot/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type BotResource struct {
+}
+
+func TestAccBot_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_bot", "test")
+	r := BotResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basicConfig(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("kind").HasValue("azurebot"),
+			),
+		},
+		data.ImportStep("developer_app_insights_api_key"),
+	})
+}
+
+func TestAccBot_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_bot", "test")
+	r := BotResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basicConfig(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("kind").HasValue("azurebot"),
+			),
+		},
+		data.ImportStep("developer_app_insights_api_key"),
+		{
+			Config: r.updateConfig(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("kind").HasValue("sdk"),
+			),
+		},
+		data.ImportStep("developer_app_insights_api_key"),
+	})
+}
+
+func TestAccBot_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_bot", "test")
+	r := BotResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.completeConfig(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("kind").HasValue("azurebot"),
+			),
+		},
+		data.ImportStep("developer_app_insights_api_key"),
+	})
+}
+
+func (t BotResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.BotServiceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Bot.BotClient.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %v", id.String(), err)
```

Comment:
We don't need to explicitly call the `String()` method
```suggestion
		return nil, fmt.Errorf("retrieving %s: %v", *id, err)
```